### ISSUE TITLE
Internal server error when user is not part of a GitHub repo

### ISSUE THREAD

**Issue by languitar** [oauth2-proxy/oauth2-proxy#2063]
When using the `--github-repo` argument to check if a users has access to a repository, the proxy returns an internal server in case the user is not part of the repository instead of an unauthorized response.

## Expected Behavior

HTTP unauthorized should be returned

## Current Behavior

500 internal server error

## Possible Solution

The Github provider seems to request the following API URL to determine if someone is a collaborator on a repo:

```
/api/v3/repos/<org>/<repo>/collaborators/<username>
```

This route returns a 404 in case this user does not exist on the repo. Instead of treating that as an unauthorized, the proxy ends up in a 500 server error.

## Steps to Reproduce (for bugs)

1.  Create a GitHub OAuth app
2.  Configure the proxy to use that app
3.  Add the `--github-repo` command line flag for a repo the user to log in has no access to
4.  Try to log in with such a user

Adding a `--github-token` as indicated in the documentation also does not help because the API would still return a 404 for the route shown before.

## Context

Seeing a 500 internal server error instead of some unauthorized message confuses users.

## Your Environment

- Version used: tested on 7.3.0 and 7.4.0

**Comment by benni336** [comment]
@languitar: Thanks for posting this. I was just fiddling with something similar and thought about making a post too.

When using `github_user(s)` with _not_ my username, I get a `500 Internal server error: missing github user` too.

I would expect an 403 unauthorized access.

Steps to reproduces:
1) run with ENV-Var: `- OAUTH2_PROXY_GITHUB_USERS=GITHUBUSERNAME` where your `GITHUBUSERNAME` is your github username. This should work.
2) restart and change `GITHUBUSERNAME` to something that does not match your username.

I want to report this behavior for `v7.2.1`

**Comment by languitar** [comment]
Still valid

**Comment by JoelSpeed** [comment]
@languitar It sounds like you've identified the bug, do you think you could raise a PR to resolve the issue?

**Comment by languitar** [comment]
Still relevant. I am not a Go person, so not easy for me to provide a viable fix.

**Comment by JoelSpeed** [comment]
Still waiting on a GitHub provider user to create a PR for this

**Comment by laszlocph** [comment]
I also see this issue.
